### ISSUE TITLE
BUILD file broke Makefile

### ISSUE THREAD

**Issue by lblackstone** [coreos/tectonic-installer#2592]
## Is this a BUG REPORT or FEATURE REQUEST?

BUG REPORT

## Versions

* Tectonic version ([release](https://github.com/coreos/tectonic-installer/releases) or commit hash): 682ebae2f50727a1a2baddce4a911d719edaa68b
* Terraform version (`terraform version`): n/a
* Platform (aws|azure|openstack|metal): n/a

## What happened?
`BUILD` file added in #2221 stomps on the `build/` directory used for cluster configs.

## What you expected to happen?
`build/` directory containing cluster configs still accessible, and `make` commands work.

## How to reproduce it (as minimally and precisely as possible)?
Run `CLUSTER=foo PLATFORM=openstack/neutron make localconfig`

```
$ CLUSTER=foo PLATFORM=openstack/neutron make localconfig
Using build directory [.../go/src/github.com/coreos/tectonic-installer/build/foo]
mkdir -p .../go/src/github.com/coreos/tectonic-installer/build/foo
mkdir: .../go/src/github.com/coreos/tectonic-installer/build/foo: Not a directory
make: *** [localconfig] Error 1
```

## Anything else we need to know?
I'm using a Mac, which has a case-insensitive file system. Haven't tested to see if this works on a Linux box.

/cc @alexsomesan

**Comment by squat** [comment]
Tested this locally and make still works on Linux. Need to investigate Mac.

**Comment by squat** [comment]
@lblackstone what MacOS version and filesystem are you running? We are planning on swapping out the makefile build system in the short term, so depending on how many people are affected, we may choose to defer a direct fix for this issue in favor of switching to Bazel.

**Comment by lblackstone** [comment]
@squat I'm running `macOS 10.13.2 (High Sierra)` with a `APFS (Encrypted)` filesystem.

I'm guessing it has to do with the case-insensitive filesystem because the `build/` folder showed up again when I deleted the Bazel `BUILD` file from the root directory.

**Comment by squat** [comment]
Yes I see. From a little investigation, it looks like APFS comes in both case-sensitive and case-insensitive flavors, and that case-insensitive is the default on macOS [0] :/

[0] https://developer.apple.com/library/content/documentation/FileManagement/Conceptual/APFS_Guide/FAQ/FAQ.html

**Comment by squat** [comment]
Fixed in #2610
